cpp/src/replace/replace.cu

```diff
@@ -247,7 +409,91 @@ struct replace_kernel_forwarder {
                                            cudf::column_view const& replacement_values,
                                            rmm::mr::device_memory_resource* mr,
                                            cudaStream_t stream = 0) {
-    CUDF_FAIL("Non fixed-width types are not supported.");
+    rmm::device_scalar<cudf::size_type> valid_counter(0);
+    cudf::size_type *valid_count = valid_counter.data();
+
+    auto replace_first = replace_strings_first_pass<true, false>;
+    auto replace_second = replace_strings_second_pass<true, false>;
+    if (input_col.has_nulls()) {
+      if (replacement_values.has_nulls()) {
+        replace_first = replace_strings_first_pass<true, true>;
+        replace_second = replace_strings_second_pass<true,true>;
+      }
+    } else {
+      if (replacement_values.has_nulls()) {
+        replace_first = replace_strings_first_pass<false, true>;
+        replace_second = replace_strings_second_pass<false, true>;
+      } else {
+        replace_first = replace_strings_first_pass<false, false>;
+        replace_second = replace_strings_second_pass<false, false>;
+      }
+    }
+
+    // Create new offsets column to use in kernel
+    std::unique_ptr<cudf::column> offsets = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32),
```

Comment:
This column's memory is part of the output, so it should use the passed-in MR.
```suggestion
    std::unique_ptr<cudf::column> offsets = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32),
     input_col.size() + 1, mask_state::UNALLOCATED, stream, mr);
```

Reply:
I note this is missing in the fixed-width version as well.

Reply:
Changed.